### ISSUE TITLE
"Error: Invalid dynamic for_each value" when upgrading version from 2.1.0 to >= 2.1.1

### ISSUE THREAD

**Issue by bertoneone** [terraform-iaac/terraform-helm-nginx-controller#17]
I get this error when I change the version from 2.1.0 to 2.1.1 (or 2.2.0):
```
Error: Invalid dynamic for_each value
│ 
│   on .terraform/modules/k8s_resources.nginx_controller/main.tf line 69, in resource "helm_release" "application":
│   69:     for_each = var.additional_set
│ 
│ Cannot use a set of object value in for_each. An iterable collection is
│ required.
╵
Operation failed: failed running terraform plan (exit 1)
```

A simplified code to replicate the issue with (terraform plan):
```
provider "aws" {
  region = "eu-west-1"
}

# required module version and your backend
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 5.0.1"
    }
  }

  required_version = ">= 0.14.0"
}

resource "aws_ssm_parameter" "test_key_wr_1" {
  name   = "/terraform/fs1/test_key_1"
  type   = "String"
  value  = "paperino"
}

data "aws_ssm_parameter" "test_key_rd_1" {
  name = "/terraform/fs1/test_key_1"
  depends_on = [ aws_ssm_parameter.test_key_wr_1 ]
}

output "ssm_stored_value_1" {
  value     = data.aws_ssm_parameter.test_key_rd_1.value
  sensitive = true
}

locals {
  nginx_controller_values = [
    {
      name  = "controller.service.annotations.service\\.beta\\.kubernetes\\.io/aws-load-balancer-ssl-cert"
      value = data.aws_ssm_parameter.test_key_rd_1.value
      type  = "string"
    },
    {
      name  = "controller.service.annotations.service\\.beta\\.kubernetes\\.io/aws-load-balancer-ssl-ports"
      value = "443"
      type  = "string"
    }
  ]
}

module "nginx_controller" {
  source  = "terraform-iaac/nginx-controller/helm"
  version = "2.2.0"

  namespace = "pippo"

  additional_set = local.nginx_controller_values
}
```

- Hardcoding a string with the arn makes it work
- casting `tostring(data.aws_ssm_parameter.test_key_rd_1.value)` here works but it seems not stable (I get the error in the real code with the same exact structure)
- aws_ssm is used to keep the snippet similar to what I have (I am importing one value from ssm and that makes it fail; that same value is written elsewhere and used in other parts without errors)

Everything on my side breaks after version 2.1.0, before that, it works fine.

**Comment by bertoneone** [comment]
Turns out that the problem in my code arises after commit [a15dd3c](https://github.com/terraform-iaac/terraform-helm-nginx-controller/compare/v2.1.0...v2.1.1) and is caused by the injection of a sensitive string variable in one of the objects in the additional_set variable that now has a type declaration.

It seems to me a weird behavior (using string conversion sometimes makes it work) and a difficult to understand error but that is not caused by this module. 

Using nonsensitive() fixes everything.

**Comment by bohdantverdyi** [comment]
Fixed, use v2.2.1 version please.
